src/fheroes2/dialog/dialog_quickinfo.cpp

```diff
@@ -262,6 +262,34 @@ namespace
         return str;
     }
 
+    std::string showTreeOfKnowledgeInfo( const Maps::Tiles & tile, const bool isVisited )
+    {
+        const MP2::MapObjectType objectType = tile.GetObject( false );
+        std::string str = MP2::StringObject( objectType );
+
+        if ( isVisited ) {
+            const Funds & payment = getTreeOfKnowledgeRequirement( tile );
+            str.append( "\n\n(" );
+            if ( payment.GetValidItemsCount() == 0 ) {
+                str.append( _( "treeOfKnowledge|free" ) );
+            }
+            else {
+                const auto rc = payment.getFirstValidResource();
+                str.append( std::to_string( rc.second ) + " " );
```

Comment:
Let's do this way:
```cpp
str.append( std::to_string( rc.second ) );
str += ' ';
```
This should be faster.

Reply:
Hi, @zenseii, in the last commit the `str += ' ';` was added, so we should remove this inserted space from the upper line :)
```suggestion
                str.append( std::to_string( rc.second ) );
```